test/parallel/test-common-expect-warning.js

```diff
@@ -36,9 +36,11 @@ if (process.argv[2] !== 'child') {
     child.stderr.on('data', (data) => {
       stderr += data;
     });
+    child.stderr.on('end', common.mustCall(() => {
+      assert.match(stderr, /Unexpected extra warning received/);
+    }));
     child.on('exit', common.mustCall((status) => {
       assert.notStrictEqual(status, 0);
-      assert.match(stderr, /Unexpected extra warning received/);
```

Comment:
Isn't the issue that the output is truncated?

Reply:
It could be, yes, but the process exits without errors so my guess is that data is written but read too late. We should check if this changes anything on arm64 where, according to CI results, the issue reproduces more frequently.